src/UnitTests/Core/Impl/Paths.cs

```diff
@@ -3,22 +3,43 @@
 
 using System;
 using System.IO;
+using System.Text;
 using Microsoft.Common.Core;
-using Microsoft.Win32;
-using static System.FormattableString;
 
 namespace Microsoft.UnitTests.Core {
     public static class Paths {
-        private static Lazy<string> VsRootLazy { get; } = Lazy.Create(() => (string)Registry.GetValue(Invariant($"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\VisualStudio\\{Toolset.Version}"), "InstallDir", string.Empty));
-        private static Lazy<string> VsPrivateAssembliesLazy { get; } = Lazy.Create(() => Path.Combine(VsRootLazy.Value, @"PrivateAssemblies\"));
-        private static Lazy<string> VsCommonExtensionsLazy { get; } = Lazy.Create(() => Path.Combine(VsRootLazy.Value, @"CommonExtensions\"));
+        private static string _vsRoot;
+        public static string VsRoot {
+            get {
+                if (_vsRoot == null) {
+#if VS14
+                    _vsRoot = (string)Registry.GetValue(Invariant($"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\VisualStudio\\{Toolset.Version}"), "InstallDir", string.Empty);
```

Comment:
We should either use `Lazy` or wrap it with `lock`, cause this field is accessed in tests from multiple threads.
I'd prefer `Lazy`.